src/main/java/uk/gov/hmcts/reform/pip/rules/repository/InMemoryRepository.java

```diff
@@ -71,5 +106,39 @@
     }
 
 
+    /**
+     * Method to retrieve a court hearings using court ID.
+     * @param courtId The ID of the court.
+     * @return The court hearings to return, if found.
+     */
+    public Optional<CourtHearings> getCourtHearings(Integer courtId)
+    {
+        CourtHearings court = this.courtHearings.stream()
+            .filter(c -> courtId.equals(c.getCourtId()))
+            .findAny()
+            .orElse(null);
+
+        List<Hearing> listHearings = this.listHearings.stream()
+            .filter(h -> h.getCourtId() == courtId)
+            .collect(Collectors.toList());
+
+        if (court != null) {
+            court.setHearingList(listHearings);
+            return Optional.of(court);
+        }
+        else {
+            return Optional.empty();
+        }
+
+
+    }
+
+    public List<CourtHearings> getCourtHearings() {
+        return this.courtHearings;
+    }
+
+    private List<CourtHearings> courtHearings;
```

Comment:
assign class variables at the top please

Reply:
class variables moved to the top